### PR TITLE
Action Cable: move channel_name to Channel.broadcasting_for

### PR DESCRIPTION
**Pull request by palkan**
### Summary

That main purpose of this PR is to make the following example _testable_:

```ruby
class ChatRelayJob < ApplicationJob
  def perform_later(room, msg)
    ChatChannel.broadcast_to room, message: msg
  end
end
```

To test this functionality we need to know the underlying stream name
(to use `assert_broadcasts`), which relies on `channel_name`.

We had to use the following code:

```ruby
assert_broadcasts(ChatChannel.broadcasting_for([ChatChannel.channel_name, room]), 1) do
  ChatRelayJob.perform_now
end
```

The problem with this approach is that we use _internal_ API: we shouldn't care about `channel_name` prefix in our code (and how should developers know that it's even used?).

With this commit we could re-write the test as following:

```ruby
 assert_broadcasts(ChatChannel.broadcasting_for(room), 1) do
   ChatRelayJob.perform_now
 end
```

IMO, this refactoring also makes the `broadcasting_for` method much easier to use _internally_, since we should't think about adding a `channel_name` every time we use it (like we did in `channel/streams`).

This PR also adds `Channel#broadcast_to` (we had `#steam_for` but hadn't a shorter syntax for broadcasting for some reason).

**NOTE:** in [`action-cable-testing`](https://github.com/palkan/action-cable-testing) we handled this differently: by adding `channel: ...` option to assertions; but that was a temporary hack since we couldn't change the Rails API.

### CODE REVIEW

File: actioncable/lib/action_cable/channel/broadcasting.rb [hunk 1]
```diff
@@ -7,22 +7,26 @@ module Channel
...
+      delegate :broadcasting_for, :broadcast_to, to: :class
 
       module ClassMethods
         # Broadcast a hash to a unique broadcasting for this <tt>model</tt> in this channel.
         def broadcast_to(model, message)
-          ActionCable.server.broadcast(broadcasting_for([ channel_name, model ]), message)
+          ActionCable.server.broadcast(broadcasting_for(model), message)
         end
 
         def broadcasting_for(model) #:nodoc:
```
Comment (georgeclaghorn) [inline actioncable/lib/action_cable/channel/broadcasting.rb]: If this method is intended to be used in app tests, it should be documented.

File: actioncable/lib/action_cable/channel/broadcasting.rb [hunk 2]
```diff
@@ -7,22 +7,26 @@ module Channel
...
+          stringify_broadcasting([ channel_name, model ])
+        end
+
+        def stringify_broadcasting(object)
```
Comment (georgeclaghorn) [inline actioncable/lib/action_cable/channel/broadcasting.rb]: This method should be `#:nodoc`ed or made private. It doesn’t seem to be used elsewhere, so preferably the latter.

File: actioncable/lib/action_cable/channel/broadcasting.rb (hunk 1 above)
Comment (palkan) [reply actioncable/lib/action_cable/channel/broadcasting.rb]: Done.
Added a note to the change log.

File: actioncable/lib/action_cable/channel/broadcasting.rb (hunk 2 above)
Comment (kaspth) [reply actioncable/lib/action_cable/channel/broadcasting.rb]: Let’s also rename to serialize_broadcasting. Stringify sounds JS-like.

Comment (palkan) [reply actioncable/lib/action_cable/channel/broadcasting.rb]: Sure.

> Stringify sounds JS-like.

Yep) I didn't like it either)

Thanks!

File: actioncable/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,28 @@
+*   Add `Channel::Base#broadcast_to`.
+
+    You can now call `broadcast_to` within a channel action, which equals to
+    the `self.class.broadcast_to`.
+
+    *Vladimir Dementyev*
+
+*   Make `Channel::Base.broadcasting_for` a public API.
+
+    This change introduces a breaking change to internal Action Cable API:
```
Comment (kaspth) [inline actioncable/CHANGELOG.md]: A change log entry is only for public API. Just remove this “before”.

File: actioncable/lib/action_cable/channel/broadcasting.rb [hunk 4]
```diff
@@ -7,22 +7,27 @@ module Channel
...
 
       module ClassMethods
         # Broadcast a hash to a unique broadcasting for this <tt>model</tt> in this channel.
         def broadcast_to(model, message)
-          ActionCable.server.broadcast(broadcasting_for([ channel_name, model ]), message)
+          ActionCable.server.broadcast(broadcasting_for(model), message)
         end
 
-        def broadcasting_for(model) #:nodoc:
+        # Returns a unique broadcasting identifier for this <tt>model</tt> in this channel.
```
Comment (kaspth) [inline actioncable/lib/action_cable/channel/broadcasting.rb]: Now that this is public API, can we show an example?

File: guides/source/testing.md [hunk 5]
```diff
@@ -1837,6 +1837,33 @@ class ProductTest < ActionCable::TestCase
...
+end
+
+# test/jobs/chat_relay_job_test.rb
+require 'test_helper'
+
+class ChatRelayJobTest < ActiveJob::TestCase
+  include ActionCable::TestHelper
+
+  test "broadcast message to room" do
+    room = rooms[:all]
```
[the remark this answers is not in the document]
Comment (palkan) [reply guides/source/testing.md]: Oops!
Fixed in other related docs/guides too.

**Comment by kaspth** [comment]
Thanks! I just realised I should have asked you to squash your commits down to 1, oh well, next time :D

**Pull request merged**